From 25aa3304aaf55a05726b7f6fa7cc9df0dea83fe2 Mon Sep 17 00:00:00 2001
From: Fabien Dessenne <fabien.dessenne@st.com>
Date: Thu, 10 Nov 2016 15:35:10 +0100
Subject: [PATCH 59/61] compositor-st: drmModeSetPlane error management

If drmModeSetPlane() fails, do not wait for vblank, and mark the sprite
as unused.
This fixes a SIGSEGV regression introduced by "compositor-st: forbid
sprite on several crtc".

Change-Id: Ic2f320f34aa2fd032595a26cba3d5499fda49fdb
Signed-off-by: Fabien Dessenne <fabien.dessenne@st.com>
Reviewed-on: https://gerrit.st.com/60249
Reviewed-by: Romuald JEANNE <romuald.jeanne@st.com>
Tested-by: Romuald JEANNE <romuald.jeanne@st.com>
---
 libweston/compositor-st.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libweston/compositor-st.c b/libweston/compositor-st.c
index 93a8664..f0d93eb 100644
--- a/libweston/compositor-st.c
+++ b/libweston/compositor-st.c
@@ -868,6 +868,9 @@ drm_output_repaint(struct weston_output *output_base,
 			weston_log(" setplane failed: %d: %s\n",
 				ret, strerror(errno));
 			s->output = NULL;
+			s->next = NULL;
+			s->vblank_pending = false;
+			continue;
 		} else {
 			s->vblank_pending = true;
 		}
@@ -892,6 +895,9 @@ drm_output_repaint(struct weston_output *output_base,
 			weston_log(" vblank event request failed: %d: %s\n",
 				ret, strerror(errno));
 			s->output = NULL;
+			s->next = NULL;
+			s->vblank_pending = false;
+			continue;
 		}
 
 		output->vblank_pending++;
-- 
2.7.4

